sematic/types/registry.py

```diff
@@ -294,6 +294,10 @@ def validate_type_annotation(*types: TypeAnnotation) -> None:
     """
 
     def assert_supported(type_):
+        if type_ is Any:
+            raise TypeError(
+                "'Any' is not a Sematic-supported type. Use 'object' instead."
```

Comment:
I think it's because (a) the semantics of `Any` are more complicated than `object` and (b) `object` can be used with `isinstance` and `issubclass` in the ways you'd like and expect, while `Any` cannot. These latter are necessary not just for *casting* but for *compatibility check*. Ex:

```python
@func
def producer() -> int:
    return 42

@func
def consumer(x: object) -> str:
    return str(x)

@func
def pipeline() -> str
    return consumer(producer)
```

Even if we don't *cast* `42` to `object` before passing it to `consumer`, we will still want to check that `consumer` can *receive* it. This will require an `isinstance`. We will also want to check that it's valid to use `consumer` to accept the result of `producer`, which will require an `issubclass`. Again, `object` works more naturally here. So I don't see this changing even when we remove casting.